elasticdl/python/worker/worker.py

```diff
@@ -198,6 +209,25 @@ def get_model(self, version, method):
             self._non_embed_vars[tensor.name].assign(tensor.to_ndarray())
         self._model_version = model.version
 
+    def get_model_from_ps(self, version, method):
+        model_version = -1
+        for stub in self._ps_stubs:
+            req = empty_pb2.Empty()
+            res = stub.pull_variable(req)
+
+            for tensor_pb in res.model.param:
+                tensor = Tensor.from_tensor_pb(tensor_pb)
+                self._non_embed_vars[tensor.name].assign(tensor.to_ndarray())
+
+            model_version = max(model_version, res.model.version)
+        self._model_version = model_version
+
+    def get_model(self, version, method):
+        if self._use_multi_ps:
+            self.get_model_from_ps(version, method)
+        else:
+            self.get_model_from_master(version, method)
```

Comment:
Maybe `get_non_embed_vars_from_ps` and `get_non_embed_vars_from_master` ? And the tensor assignment can be replicated in `get_model`.
```python
def get_model(self, version, metho):
     ...
     for tensor_pb in params:
         tensor = Tensor.from_tensor_pb(tensor_pb)
         self._non_embed_vars[tensor.name].assign(tensor.to_ndarray())
```
I also wonder whether we need to abstract clients to communicate with the Master or the PS. For example, the worker gets tasks by the master client and gets model pararms by PS client. Now, the communication operators are all in worker.py which may be not very clean.

Reply:
The same reason, `get_model_from_master` will be removed later. We keep compatibility with master-worker architect now.  So, I just keep the master related code unchanged.

For the communication client abstraction, I think it does make sense. A worker has two clients:
- pull parameters/push gradients communication with PS
- get task/report metrics communication with master

We could consider cleaning worker code in next PRs.